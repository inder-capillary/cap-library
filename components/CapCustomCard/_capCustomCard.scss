@import "../styles/_variables.scss";

$cap-custom-card-list: cap-custom-card-list;
$cap-custom-card: cap-custom-card;

.ant-card.#{$cap-custom-card} {
  border-radius: 4px;
  display: inline-block;
  &.remove-border-radius{
    border-radius: unset;
  }
  .ant-card-head {
    border-bottom: none;
  }
  .ant-card-extra {
    padding: 0;
    line-height: 16px;
  }
  .ant-card-head-title {
    font-size: $FONT_SIZE_M;
    margin-right: $CAP_SPACE_08;
  }
  .ant-card-actions {
    display: flex;
    align-items: center;
    background-color: $CAP_WHITE;
    padding-left: $CAP_SPACE_12;
    padding-right: $CAP_SPACE_12;
    min-height: 49px;
    li {
      margin: $CAP_SPACE_04 0;
      border-right: none;
      float: none;
    }
  }
  .ant-card-actions > li > span {
    display: block;
    & > .anticon {
      font-size: $FONT_SIZE_VL;
    }
    &:hover {
      color: inherit;
    }
  }

  .ant-card-meta-description {
    font-size: $FONT_SIZE_S;
    color: #5e6c84;
  }
  .hover-content {
    display: none;
  }

  &.has-hover-option:hover {
    .ant-card-meta {
      opacity: 0.2;
    }
    .hover-content {
      display: block;
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -40%);
    }
  }
}

.ant-card.#{$cap-custom-card}.SMS {
  .ant-card-body {
    padding-bottom: 0;
    padding-top: $CAP_SPACE_04;
    height: 129px;
    display: block;
    display: -webkit-box;
    -webkit-line-clamp: 7;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .ant-card-body {
    /*! autoprefixer: off */
    -webkit-box-orient: vertical;
  }
}

.ant-card.#{$cap-custom-card}.CALL_TASK {
  .ant-card-body {
    padding-bottom: 0;
    padding-top: $CAP_SPACE_04;
    height: 129px;
  }
}

.ant-card.#{$cap-custom-card}.MOBILEPUSH, .ant-card.#{$cap-custom-card}.WECHAT {
  .ant-card-body {
    padding-bottom: 0;
    padding-top: $CAP_SPACE_04;
    padding-left: $CAP_SPACE_16;
    padding-right: $CAP_SPACE_16;
    background-color: $CAP_G09;
    height: 279px;
    display: block;
    display: -webkit-box;
    -webkit-line-clamp: 7;
    overflow: hidden;
    text-overflow: ellipsis;
    .app-name{
      font-size: $FONT_SIZE_S;
      padding-left: $CAP_SPACE_12;
      padding-top: $CAP_SPACE_04;
      background-color: $CAP_WHITE;
      margin-top: $CAP_SPACE_16;
    }
    .ant-card-meta {
      background-color: $CAP_WHITE;
      padding: $CAP_SPACE_12;
      border-radius: 6px;
    }
    .ios-action-buttons{
      margin-top: $CAP_SPACE_24;
      .ios-action-label{
        width: 100%;
        height: $CAP_SPACE_40;
        line-height: $CAP_SPACE_24;
        background-color: $CAP_WHITE;
        padding-top: $CAP_SPACE_12;
        text-align: center;
        &:first-child{
          border-top-left-radius: 6px;
          border-top-right-radius: 6px;
          border-bottom: 1px solid $CAP_G07;
        }
        &:last-child{
          border-bottom-left-radius: 6px;
          border-bottom-right-radius: 6px;
        }
      }
    }
    .wechat-card-body {
      margin-top: $CAP_SPACE_12;
      margin-bottom: $CAP_SPACE_16;
      background-color: $CAP_WHITE;
      height: 247px;
      border-radius: $CAP_SPACE_04;
      .map-template {
        padding: $CAP_SPACE_12 $CAP_SPACE_16;
      }
      .richmedia-template {
        .richmedia-image-container {
          width: 100%;
          height: 136px;
          position: relative;
          img {
            border-top-left-radius: 4px;
            border-top-right-radius: 4px;
          }
        }
        .name-container-with-abstract {
          padding: $CAP_SPACE_12;
          height: 68px;
        }
        .name-container {
          position: absolute;
          bottom: 0px;
          height: 44px;
          padding-top: 16px;
          padding-left: 12px;
          color: $CAP_WHITE;
          opacity: 0.7;
          width: 100%;
          background-image: linear-gradient(to top,black 0% ,transparent 100%);
        }
        .additional-richmedia-container {
          display: flex;
          justify-content: space-between;
          margin: 8px;
          padding-bottom: 8px;
          img {
            border-radius: 4px;
          }
        }
        .additional-richmedia-container:not(:last-child) {
          border-bottom: 1px solid #dfe2e7;
        }
        div {
          text-overflow: ellipsis;
          overflow: hidden;
          white-space: nowrap;
        }
      }
    }
  }

  .ant-card-body {
    /*! autoprefixer: off */
    -webkit-box-orient: vertical;
  }
  .preview-overflow-wrap-fields {
    text-overflow: ellipsis;
    overflow: hidden;
    overflow-wrap: break-word;
    display: -webkit-box !important;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 2;
    text-align: left;
    width: 200px;
  }
}

.facebook-card-body {

  .ant-skeleton-with-avatar {
    .ant-skeleton-content {
      .ant-skeleton-title {
        border-radius: 4px;
        margin-top: 4px;
      }
    }
  }

  .ant-skeleton-paragraph {
    margin-left: -$CAP_SPACE_52;

    li {
      border-radius: 4px;

      &:first-child {
        margin: -$CAP_SPACE_12 $CAP_SPACE_52 $CAP_SPACE_20;
      }
    }
  }

  .dynamic-content-section {
    width: 100%;
    padding: 30px 44px;
    background-color: $CAP_G09;
    margin-top: $CAP_SPACE_08;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 4px;

    .centered-item {
      max-width: 130px;
      text-align: center;
    }
  }
}

.ant-card.#{$cap-custom-card}.EMAIL {
  &.no-image{
    .ant-card-body{
      padding-bottom: 0;
      padding-top: $CAP_SPACE_04;
      height: 129px;
      display: block;
      display: -webkit-box;
      -webkit-line-clamp: 7;
      overflow: hidden;
      text-overflow: ellipsis;
    }  
  }
  }

.#{$cap-custom-card-list} {
  .ant-card.#{$cap-custom-card} {
    display: block;
  }
  .#{$cap-custom-card-list}-row {
    display: flex;
    flex-flow: row wrap;
    justify-content: flex-start;
    .#{$cap-custom-card-list}-col {
      margin-top: $CAP_SPACE_08;
      width: 276px;
      margin-right: 12px;
    }
  }
}
