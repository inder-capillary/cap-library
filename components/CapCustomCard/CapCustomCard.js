/**
*
* CapCard
*
*/

import React from 'react';
import './_capCustomCard.scss';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import { Card, Avatar } from "antd";
import get from 'lodash/get';
import isEmpty from 'lodash/isEmpty';
import CapHeader from "../CapHeader";
import CapLink from "../CapLink";
const { Meta } = Card;

const clsPrefix = "cap-custom-card";
const channels = {
  sms: 'SMS',
  email: 'EMAIL',
  mpush: "MOBILEPUSH",
};
class CapCustomCard extends React.Component {
    getMpushContent = (templateData) => {
      const mpushData = get(templateData, 'versions.base') || templateData;
      const androidData = get(mpushData, 'ANDROID') || get(mpushData, 'androidContent');
      const iosData = get(mpushData, 'IOS') || get(mpushData, 'iosContent');
      let title = get(androidData, 'title');
      let description = get(androidData, 'message');
      let actionText1 = get(androidData, 'expandableDetails.ctas[0].actionText');
      let actionText2 = get(androidData, 'expandableDetails.ctas[1].actionText');

      let hasImage = get(androidData, 'expandableDetails.style') === "BIG_PICTURE";
      let image;
      if (hasImage) {
        image = get(androidData, 'expandableDetails.image');
      }
      if (isEmpty(androidData)) {
        title = get(iosData, 'title');
        description = get(iosData, 'message');
        actionText1 = get(iosData, 'expandableDetails.ctas[0].actionLabel');
        actionText2 = get(iosData, 'expandableDetails.ctas[0].actionLabel2');
        hasImage = get(iosData, 'expandableDetails.style') === "BIG_PICTURE";
        if (hasImage) {
          image = get(iosData, 'expandableDetails.image');
        }
      }
      const preview = {};
      preview.content = (
      <>

        <CapHeader size="small" title={title} description={description} />
        {hasImage && <img src={image} alt="mpush" />}
        {!isEmpty(androidData)
         && (
           <div style={{marginTop: '8px'}}>
             {actionText1 && <CapLink title={actionText1} style={{display: 'inline-block'}} />}
             {actionText2 && <CapLink title={actionText2} style={{display: 'inline-block', marginLeft: '24px'}} />}
           </div>
         )

        }

      </>
      );
      if (isEmpty(androidData)) {
        preview.iosButtons = (
          (
            <div className="ios-action-buttons">
              <div className="ios-action-label">
                <CapLink title={actionText1} />
              </div>
              <div className="ios-action-label">
                <CapLink title={actionText2} />
              </div>
            </div>
          )
        );
      }
      return preview;
    }

    getCardContent = () => {
      const { content, url, width, height } = this.props;
      const type = this.props.type.toUpperCase();
      switch (type) {
        case channels.sms:
          return <Meta description={content} />;
        case channels.email:
          return <Meta description={<img width={width || 244} height={height || 279} src={url} alt={url} />} />;
        case channels.mpush: {
          const contentPreview = this.getMpushContent(content);
          const previewMpush = [
            <div className="app-name">
              <CapHeader size="small" title={<span style={{marginLeft: '4px'}}>App name</span>} prefix={<img src="./favicon.ico" alt="" />} />
            </div>,
            <Meta description={contentPreview.content} avatar={<Avatar icon="user" />} />,
          ];
          if (contentPreview.iosButtons) {
            previewMpush.push(contentPreview.iosButtons);
          }
          return previewMpush;
        }
        default:
          return null;
      }
    }

    render() {
      const { className, type, content, hoverOption, url, width, height, ...rest } = this.props;
      return (
        <Card
          className={ClassNames(clsPrefix, className, type, { 'has-hover-option': hoverOption })}
          {...rest}
        >
          {this.getCardContent()}
          {hoverOption && (
            <div className="hover-content">
              {hoverOption}
            </div>
          )}
        </Card>
      );
    }
}

CapCustomCard.propTypes = {
  className: PropTypes.string,
  cardList: PropTypes.array,
  type: PropTypes.string,
};

export default CapCustomCard;
