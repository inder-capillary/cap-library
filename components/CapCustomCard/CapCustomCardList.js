import React from 'react';
import classNames from 'classnames';
import CapCustomCard from './CapCustomCard';


const clsPrefix = "cap-custom-card-list";

export default class CapCustomCardList extends React.Component {
  render() {
    const { className, cardList, type } = this.props;
    return (
      <div className={classNames(clsPrefix, type, className, )}>
        {cardList && (
          <div className={classNames(`${clsPrefix}-row`)}>
            {
              cardList.map((data, i) => {
                const { dataBelowCard, cardType, cardTop, ...rest } = data;
                const key = data.key || i;
                return (
                  <div className={classNames(`${clsPrefix}-col`)}>
                    {cardTop && cardTop}
                    <CapCustomCard key={key} type={cardType || type} cardTop={cardTop} {...rest} />
                    {
                      dataBelowCard && dataBelowCard
                    }
                  </div>
                );
              })
            }
          </div>
        )}
      </div>
    );
  }
}
