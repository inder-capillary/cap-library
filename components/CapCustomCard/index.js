/**
*
* CapCard
*
*/

import CapCustomCard from './CapCustomCard';
import CapCustomCardList from './CapCustomCardList';

CapCustomCard.CapCustomCardList = CapCustomCardList;

export default CapCustomCard;
