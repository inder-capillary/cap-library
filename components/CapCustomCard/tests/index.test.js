import renderer from 'react-test-renderer';
import React from 'react'; // eslint-disable-line no-unused-vars
import CapCustomCard from '../index';

describe('<CapCustomCard />', () => {
  it('CapCustomCard renders correctly', () => {
    const tree = renderer.create(<CapCustomCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
