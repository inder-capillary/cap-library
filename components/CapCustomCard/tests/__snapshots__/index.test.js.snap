// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapCustomCard /> CapCustomCard renders correctly 1`] = `
<div
  className="ant-card cap-custom-card sms ant-card-bordered"
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-card-meta"
    />
  </div>
</div>
`;
